### PR TITLE
Add specialised view for infiltration installation

### PR DESCRIPTION
**Pull request by cymed**
## General
 - [ ] Fix a bug
 - [x] Add a feature
 - [ ] Maintenance / sustainability
 - [ ] Add Documentation

## Describe your changes

- [x] Add specialised view for infiltration installation
- [x] includes retention_body and mechanical_pretreatment.

Context : informations: retention_body and mechanical_pretreatment are classes referencing to infiltration installation. In some cantons, it is mandatory to fill out this info. This PR creates a view that can be used to fill out the necessary forms more easily

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] CI Tests are green
- [ ] The documentation is up to date with the proposed change.
- [x] My work is ready for review

## Checklist before merge
- [ ] A review has been performed
- [ ] Comments are resolved
- [ ] Documentation is ready

### CODE REVIEW

**Comment by ponceta** [comment]
@cymed we need to have more context and maybe screenshots on what we want to achieve with such an evolution.

**Comment by cymed** [comment]
retention_body and mechanical_pretreatment are classes referencing to infiltration installation. In some cantons, it is mandatory to fill out this info. This PR creates a view that can be used to fill out the necessary forms more easily

**Comment by ponceta** [comment]
@cymed what are metafunctions and why are these included here?

**Comment by cymed** [comment]
For now it is only one function that can check if all entries for a prefix are null

File: datamodel/app/create_app.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 except ImportError:
     import psycopg2 as psycopg
 
+from gep_views.vw_tww_infiltration_installation import vw_tww_infiltration_installation
```
Comment (ponceta) [inline datamodel/app/create_app.py:12]: Please move it to view.vw_tww_infiltration_installation gep_views is currently at the wrong place.

This should be reorganised in the future as :
views / vw_*
views / vw_tww_*
views / export / vw_*
views / SWMM/ vw_swmm_*
views / GWMP / vw_gwmp (gep is german) (only if we have the necessity to create some distinction)

File: datamodel/app/functions/meta_functions.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+CREATE OR REPLACE FUNCTION tww_app.check_all_nulls(rec record, prefix varchar(2),ignored_postfix text[] default Array['obj_id','identifier'])
```
Comment (ponceta) [inline datamodel/app/functions/meta_functions.sql:1]: Can we have a comment or some more explanation of what does this function exactly?

**Review by ponceta — CHANGES_REQUESTED** [review]

File: datamodel/app/functions/meta_functions.sql (hunk 2 above)
Comment (cymed) [reply datamodel/app/functions/meta_functions.sql:1]: It allows checking a record (usually NEW) on whether or not its attributes with a certain prefix are all NULL. The third argument allows whitelisting auto-populated values

File: datamodel/app/create_app.py (hunk 1 above)
Comment (cymed) [reply datamodel/app/create_app.py]: resolved in https://github.com/teksi/wastewater/pull/416/commits/91d6336daa466defef9a4acdfbb789ec52468b95

**Review by ponceta — APPROVED** [review]

**Pull request merged**